src/main/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/security/MaintainUserCheck.kt

```diff
@@ -33,6 +35,20 @@ class MaintainUserCheck(
     }
   }
 
+  @Throws(UserGroupRelationshipException::class)
+  fun ensureUserLoggedInUserRelationship(loggedInUser: String, authorities: Collection<GrantedAuthority>, user: User) {
+    // if they have maintain privileges then all good
+    if (canMaintainUsers(authorities)) {
+      return
+    }
+    // otherwise group managers must have a group in common for maintenance
+    val loggedInUserEmail = userRepository.findByUsername(loggedInUser).orElseThrow()
+    if (Sets.intersection(loggedInUserEmail.groups, user.groups).isEmpty()) {
+      // no group in common, so disallow
+      throw UserGroupRelationshipException(user.name, "User not with your groups")
+    }
+  }
+
   class UserGroupRelationshipException(val username: String, val errorCode: String) :
```

Comment:
this is just a copy and paste from auth

Reply:
do we need to handle UserGroupRelationshipException in HmppsExternalUsersApiExceptionHandler?